Add standard LDFLAGS behavior to app and python-bindings

Also move the -shared into the link line directly, as it's not an optional
linking flags (in python-binding).

Signed-off-by: Mark Hatle <mark.hatle@xilinx.com>

diff --git a/app/Makefile b/app/Makefile
index 5bed9c4..24942c3 100644
--- a/app/Makefile
+++ b/app/Makefile
@@ -1,7 +1,8 @@
 .PHONY:	clean
 
 CC ?=  gcc
-CFLAGS = -Wall -Wextra
+CFLAGS ?= -Wall -Wextra
+LDFLAGS ?= 
 LIBDIR = ../src
 INCLUDEDIR = ../include/platformstats
 SOURCES = main.c
@@ -13,7 +14,7 @@ all: build
 build: $(APPSTMPS)
 
 %.out: %.o
-	$(CC) -o platformstats main.o -L$(LIBDIR) -lplatformstats
+	$(CC) $(LDFLAGS) -o platformstats main.o -L$(LIBDIR) -lplatformstats
 
 %.o: %.c
 	$(CC) -I$(INCLUDEDIR) $(CFLAGS) -c main.c -o main.o
diff --git a/python-bindings/Makefile b/python-bindings/Makefile
index 4f816c4..8d57e6e 100644
--- a/python-bindings/Makefile
+++ b/python-bindings/Makefile
@@ -2,7 +2,8 @@
 
 #Dynamic loading for C
 CC ?= gcc
-CFLAGS ?= -fPIC -shared -Wall -Wextra
+CFLAGS ?= -fPIC -Wall -Wextra
+LDFLAGS ?= 
 LIBS = -L../src -L. -lplatformstats
 INCLUDES = -I../include/platformstats
 
@@ -26,7 +27,7 @@ $(ISRCS): $(INTERFACE)
 	$(SWIG) -python $<
 
 $(PYBIND_LIB): $(ISRCS)
-	$(CC) $(CFLAGS) $(ISRCS) -o $(PYBIND_LIB) $(LIBS) $(INCLUDES) $(PYTHON_INCLUDE)
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared $(ISRCS) -o $(PYBIND_LIB) $(LIBS) $(INCLUDES) $(PYTHON_INCLUDE)
 
 
 clean:
